gov/politeia/tlog/proposals.go

```diff
@@ -0,0 +1,596 @@
+// Copyright (c) 2019-2021, The Decred developers
+// See LICENSE for details.
+
+// Tlog package is meant to be a rewrite that adapts politeia/proposals
+// package to the new tlog backend proposals. This will replace the current
+// proposals code that deals with the git backend proposals. Investigate
+// how to take and maintain a snapshot of the git proposals DB
+package tlog
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/asdine/storm/v3"
+	"github.com/asdine/storm/v3/q"
+	pitypes "github.com/decred/dcrdata/gov/v4/politeia/types"
+	"github.com/decred/dcrdata/v6/semver"
+	commentsv1 "github.com/decred/politeia/politeiawww/api/comments/v1"
+	recordsv1 "github.com/decred/politeia/politeiawww/api/records/v1"
+	ticketvotev1 "github.com/decred/politeia/politeiawww/api/ticketvote/v1"
+	piclient "github.com/decred/politeia/politeiawww/client"
+)
+
+var (
+	// errDef defines the default error returned if the proposals db was not
+	// initialized correctly.
+	errDef = fmt.Errorf("ProposalDB was not initialized correctly")
+
+	// dbVersion is the current requirxed version of the proposals.db.
+	dbVersion = semver.NewSemver(2, 0, 0)
+)
+
+// dbinfo defines the property that holds the db version.
+const dbinfo = "_proposals.db_"
+
+// TODO: remove tlog from namings (?)
```

Comment:
Yeah, the tlog aspect seems irrelevant since afaict it's just hitting /api endpoints and not dealing with anything tlog-related, correct?

Reply:
yep correct, have to clean a lot of stuff up :)

Reply:
Cool.  Take your time.  I like the way this is progressing.  Very eager to see the go-piparser (the Pi git repo package) go away and to pull vote data even for charts from the new (tlog) Pi API.